addons/survey/tests/test_survey.py

```diff
@@ -120,53 +120,3 @@ def test_answer_validation_char_box_length(self):
             question.validate_question('valid'),
             {}
         )
-
-    @users('survey_manager')
-    def test_result_data_simple_multiple_choice(self):
-        question = self._add_question(
-            self.page_0, 'Q0', 'simple_choice',
-            labels=[{'value': 'Choice0'}, {'value': 'Choice1'}]
-        )
-        for i in range(3):
-            answer = self._add_answer(self.survey, False, email='public@example.com')
-            self._add_answer_line(
-                question, answer, random.choice(question.suggested_answer_ids.ids),
-                answer_type='suggestion', answer_fname='suggested_answer_id')
-        lines = [line.suggested_answer_id.id for line in question.user_input_line_ids]
-        answers = [{'text': label.value, 'count': lines.count(label.id), 'answer_id': label.id, 'answer_score': label.answer_score} for label in question.suggested_answer_ids]
-        prp_result = self.env['survey.survey'].prepare_result(question)['answers']
-        self.assertItemsEqual(prp_result, answers)
-
-    @users('survey_manager')
-    def test_result_data_matrix(self):
-        question = self._add_question(
-            self.page_0, 'Q0', 'matrix', matrix_subtype='simple',
-            labels=[{'value': 'Column0'}, {'value': 'Column1'}],
-            labels_2=[{'value': 'Row0'}, {'value': 'Row1'}]
-        )
-        for i in range(3):
-            answer = self._add_answer(self.survey, False, email='public@example.com')
-            self._add_answer_line(
-                question, answer, random.choice(question.suggested_answer_ids.ids),
-                answer_type='suggestion', answer_fname='suggested_answer_id', matrix_row_id=random.choice(question.matrix_row_ids.ids)
-            )
-        lines = [(line.matrix_row_id.id, line.suggested_answer_id.id) for line in question.user_input_line_ids]
-        res = {}
-        for i in product(question.matrix_row_ids.ids, question.suggested_answer_ids.ids):
-            res[i] = lines.count((i))
-        self.assertEqual(self.env['survey.survey'].prepare_result(question)['result'], res)
-
-    @users('survey_manager')
-    def test_result_data_numeric_box(self):
-        question = self._add_question(self.page_0, 'Q0', 'numerical_box')
-        num = [float(n) for n in random.sample(range(1, 100), 3)]
-        nsum = sum(num)
-        for i in range(3):
-            answer = self._add_answer(self.survey, False, email='public@example.com')
-            self._add_answer_line(question, answer, num[i])
-        exresult = {
-            'average': round((nsum / len(num)), 2), 'max': round(max(num), 2),
-            'min': round(min(num), 2), 'sum': nsum, 'most_common': Counter(num).most_common(5)}
-        result = self.env['survey.survey'].prepare_result(question)
-        for key in exresult:
-            self.assertEqual(result[key], exresult[key])
```

Comment:
Useless tests I assume?

Reply:
Related to old code generating results structure. This kind of test is unnecessary because it tests nothing (if you change something in the template, test is green but page is broken) and if you change the structure, you have to adapt the test without ensuring the results page still works.